app/src/pages/SingleContent.vue

```diff
@@ -141,11 +149,10 @@ function selectTag(parentId: Uuid) {
         </div>
     </div>
 
-    <div v-if="isLoading">
-        <LoadingSpinner />
-    </div>
+    <NotFoundPage
+        v-if="isContentNotFound || isExpiredOrScheduled || content.status == PublishStatus.Draft"
+    />
```

Comment:
From just a quick glance at this: doesn't this mean their could be a short flash of "404" before the content is shown? I haven't tested this yet though.

Reply:
This should not happen under normal circumstances as content should already be in IndexedDB when routing from from the home page.

In cases where people open a link that is routing to the article without having the document in IndexedDB yet, this might happen and we will need a loading state. But I am not sure how we would be able to determine that without having a live API which can query the API directly, so perhaps we need to add a follow-up task to look at this once we have a live API available.